presto-session-property-managers/src/test/java/io/prestosql/plugin/session/db/TestDbSessionPropertyManager.java

```diff
@@ -0,0 +1,183 @@
+    }
+
+    @BeforeClass
+    public void setup()
+    {
+        config = getH2Config();
+        daoProvider = new H2SessionPropertiesDaoProvider(config);
+        dao = daoProvider.get();
+        manager = new DbSessionPropertyManager(new TestingDbSpecsProvider(dao));
+    }
+
+    @BeforeMethod
+    public void setupTest()
+    {
+        dao.dropSessionPropertiesTable();
+        dao.dropSessionSpecsTable();
+        dao.dropSessionClientTagsTable();
+        dao.createSessionSpecsTable();
+        dao.createSessionClientTagsTable();
+        dao.createSessionPropertiesTable();
+    }
+
+    static DbSessionPropertyManagerConfig getH2Config()
+    {
+        return new DbSessionPropertyManagerConfig().setConfigDbUrl("jdbc:h2:mem:test_session_properties_specs" + System.nanoTime() + ";MODE=MySQL");
+    }
+
+    /**
+     * A basic test for session property overrides with the {@link DbSessionPropertyManager}
+     */
+    @Test
+    public void testSessionProperties()
+    {
+        dao.insertSpecRow(1, "foo.*", null, null, null, 0);
+        dao.insertSessionProperty(1, "prop_1", "val_1");
+
+        dao.insertSpecRow(2, ".*", "bar.*", null, null, 0);
+        dao.insertSessionProperty(2, "prop_2", "val_2");
+
+        SessionConfigurationContext context1 = new SessionConfigurationContext("foo123", Optional.of("src1"),
+                ImmutableSet.of(), Optional.empty(), TEST_RG);
+        Map<String, String> sessionProperties1 = manager.getSystemSessionProperties(context1);
+        assertEquals(sessionProperties1.get("prop_1"), "val_1");
+        assertFalse(sessionProperties1.containsKey("prop_2"));
+
+        SessionConfigurationContext context2 = new SessionConfigurationContext("bar123", Optional.of("bar123"),
+                ImmutableSet.of(), Optional.empty(), TEST_RG);
+        Map<String, String> sessionProperties2 = manager.getSystemSessionProperties(context2);
+        assertEquals(sessionProperties2.get("prop_2"), "val_2");
+        assertFalse(sessionProperties2.containsKey("prop_1"));
+
+        SessionConfigurationContext context3 = new SessionConfigurationContext("foo123", Optional.of("bar123"),
+                ImmutableSet.of(), Optional.empty(), TEST_RG);
+        Map<String, String> sessionProperties3 = manager.getSystemSessionProperties(context3);
+        assertEquals(sessionProperties3.get("prop_1"), "val_1");
+        assertEquals(sessionProperties3.get("prop_2"), "val_2");
+
+        SessionConfigurationContext context4 = new SessionConfigurationContext("abc", Optional.empty(), ImmutableSet.of(), Optional.empty(), TEST_RG);
+        Map<String, String> sessionProperties4 = manager.getSystemSessionProperties(context4);
+        assertFalse(sessionProperties4.containsKey("prop_1"));
+        assertFalse(sessionProperties4.containsKey("prop_2"));
+    }
+
+    /**
+     * Test {@link DbSessionPropertyManager#getSystemSessionProperties} after unsuccessful reloading
+     */
+    @Test
+    public void testReloads()
```

Comment:
@electrum the refresh kind of happens implicitly, since `TestingDbSpecsProvider` invokes `loadingFunction` with every invocation of `getSessionProperties()`.

Reply:
I agree with your point that using `TestingDbSpecsProvider` does not test the actual code path in `RefreshingDbSpecsProvider`. we need to use the `RefreshingDbSpecsProvider` in the tests without scheduling the periodic refresh in the initialization.

Reply:
using `RefreshingDbSpecsProvider` in `TestDbSessionPropertyManager` now. 
the integration test still uses the Testing implementation.